src/redux/uniqueTokens.ts

```diff
@@ -277,30 +279,7 @@ export const fetchUniqueTokens = (showcaseAddress?: string) => async (
 
       // If there are any "unknown" ENS names, fallback to the ENS
       // metadata service.
-      newPageResults = await Promise.all(
-        newPageResults.map(async (token: UniqueAsset) => {
-          const isENS =
-            token?.asset_contract?.address?.toLowerCase() ===
-            ENS_NFT_CONTRACT_ADDRESS.toLowerCase();
-          if (isENS && !token.uniqueId.includes('.eth')) {
-            try {
-              const { name, image_url } = await fetchMetadata({
-                tokenId: token.id,
-              });
-              return {
-                ...token,
-                image_preview_url: image_url,
-                image_url,
-                name,
-                uniqueId: name,
-              };
-            } catch {
-              return token;
-            }
-          }
-          return token;
-        })
-      );
+      newPageResults = await applyENSMetadataFallbackToTokens(newPageResults);
```

Comment:
wondering we should filter ENS here just for readability, this is taking all nfts right? even if there aren't ENS

Reply:
Won't it need to take all NFTs anyway too check if it is ENS or not?

Reply:
I thought the "apply fallback" terminology was good enough, but can just do the `.map` here instead